client/src/main/java/io/pravega/client/ClientConfig.java

```diff
@@ -230,6 +244,20 @@ public ClientConfig build() {
             if (controllerURI == null) {
                 controllerURI = URI.create("tcp://localhost:9090");
             }
+            String scheme = controllerURI.getScheme();
+            if (!SCHEME_DISCOVER.equals(scheme) && !SCHEME_DISCOVER_TLS.equals(scheme) &&
+                    !SCHEME_DIRECT.equals(scheme) && !SCHEME_DIRECT_SSL.equals(scheme)
+                    && !SCHEME_DIRECT_TLS.equals(scheme)) {
+                // If controllerURI is kind of <HOST>:<PORT> then will add TCP as default. 
+                // Otherwise URI might have some other Unsupported Scheme like http, https, ftp etc. in those cases will return null. 
+                String uri = controllerURI.toString();
+                if (uri.toString().split(":").length == 2) {
+                    controllerURI = URI.create("tcp://" + uri);
+                } else {
+                    log.warn("Unsupported URI: " + uri);
+                    return null; 
```

Comment:
This doesn't seem like the right way to handle this. It seems like this should throw IllegalArgumentException

Reply:
Added Illegal Argument Exception.